### DISCUSSION TITLE
Lung tumor segmentation

### DISCUSSION THREAD

**Discussion by Meddebma** [Project-MONAI/MONAI#2931] [Ideas]
Hi, 

I am trying to implement the 3d-segmentation pipeline with the Lung_Dataset of MSD, and I was wondering if there are transforms specific to Lung tumor segmentation, I combined transforms of the registration and the challenge tutorials
here are the transforms I used:

train_transforms = Compose(
    [
        LoadImaged(
            keys=["image", "label"]
        ),
        AddChanneld(
            keys=["image", "label"]
        ),
        Orientationd(keys=["image", "label"], axcodes="LPS"),
        Spacingd(keys=["image", "label"], pixdim=(1.25, 1.25, 5.0), mode=("bilinear", "nearest")[: len(["image", "label"])]),
        ScaleIntensityRanged(
            keys=["image", "label"],
            a_min=-285, a_max=3770, b_min=0.0, b_max=1.0, clip=True,
        ),
        SpatialPadd(keys=["image", "label"], spatial_size=(192, 192, -1), mode="reflect"),    
        RandAffined(
            keys=["image", "label"],
            mode=('bilinear', 'nearest'),
            prob=1.0, spatial_size=(192, 192, 208),
            rotate_range=(0, 0, np.pi / 15), scale_range=(0.1, 0.1, 0.1)
        ),
        RandCropByPosNegLabeld(keys=["image", "label"], label_key=["label"], spatial_size=(192, 192, 16), num_samples=3),
        Resized(
            keys=["image", "label"],
            mode=('trilinear', 'nearest'),
            align_corners=(True, None),
            spatial_size=(96, 96, 104)
        ),
        EnsureTyped(
            keys=["image", "label"]
        ),
    ])

When I was loading the data, there was an error implementing RandCropByPosNegLabel, does it have something to do with the spacial size? Which transforms should I keep for this task? thank you very much!

**Comment by Nic-Ma** [comment]
Hi @wyli @ericspod @dongyang0122 ,

Could you please help share some training experience with `Lung` data?

Thanks in advance.

**Comment by dongyang0122** [comment]
Hi @Meddebma , you may use the following tutorial for MSD tasks. That would a good starting point.

https://github.com/Project-MONAI/tutorials/tree/master/modules/dynunet_pipeline
